src/utilities/googleAnalytics.ts

```diff
@@ -38,3 +47,53 @@ export const getNotionalActionFromHintState = (hint: SwapFormSubmitButtonHintSta
             return "";
     }
 }
+
+export const pushEvent = (eventName: string, eventValue: string | number, sessionId?: string, amm?: AugmentedAMM, agent?: string) => {
```

Comment:
eventName should a string literal type. For example we should limit what type of events we need to push.

```suggestion
type AnalyticEvents = 'a' | 'b' | 'c'...;

export const pushEvent = (eventName: AnalyticEvents, eventValue: string | number, sessionId?: string, amm?: AugmentedAMM, agent?: string) => {
```

Reply:
created two new types `DataLayerEventPayload` and `TxEventPayload` to be used as parameters for the `pushEvent` function. This funct is now general, used for tx events, input change events, title change etc... resolved 9bb0958572bdde864d23200e7f12fd6c19910057